Use generated sysdown() function

ccollins476ad opened the pull request:
NOTE: This commit requires a newt tool update (a38614ebc22959231ad9597c24467e34ed0e5068).

Upon application-triggered restart, execute each package's shutdown subprocedure.  This feature is discussed here:
https://lists.apache.org/thread.html/ebd53c9b56230184650ea57e5534d5bcfbfc170b97547a1a26f100b5@%3Cdev.mynewt.apache.org%3E

mkiiskila commented on sys/sysdown/src/sysdown.c:
```diff
@@ -0,0 +1,169 @@
+    assert(rc == 0);
+
+    /* Call each configured sysdown callback. */
+    for (i = 0; sysdown_cbs[i] != NULL; i++) {
```
Could it be sizeof(sysdown_cbs) / sizeof(sysdown_cbs[0])? Would save 4 bytes of text, if sysdown_cbs[] we're not terminated by a NULL...

ccollins476ad replied:
The reason I null-terminated the array is to handle the case where there aren't any sysdown procedures (a zero-length array being undefined).  gcc does allow zero-length arrays, but I didn't think four bytes' savings justified using an extension.

mkiiskila replied:
That's fair. Agreed.

mkiiskila commented on sys/sysdown/src/sysdown.c:
```diff
@@ -0,0 +1,169 @@
+            break;
+
+        default:
+            return SYS_EUNKNOWN;
```
I'm wondering if it would be better to continue calling sysdown() routines, and proceed with the shutdown?

ccollins476ad replied:
Yes, I agree.  It doesn't make sense for sysdown to just stop like this.

ccollins476ad commented on sys/sysdown/src/sysdown.c:
```diff
@@ -0,0 +1,169 @@
+int
+sysdown(int reason, sysdown_complete_fn *complete_cb, void *arg)
+{
+    return 0;
```
Yes, good catch.

ccollins476ad commented:
Thanks, @mkiiskila.  I believe I have addressed all your comments as follows:
1. If newt doesn't support the sysdown feature, `sysdown()` just performs a system reset.
2. If a sysdown subprocedure returns an unknown error code, pretend it returned `SYSDOWN_COMPLETE` (i.e., continue shutting down).

Also, I changed the syscfg setting name from `SYSDOWN` to `NEWT_FEATURE_SYSDOWN`.  I think we can avoid some confusion in the future by clearly identifying settings that newt injected by itself.
